spec/data-model.md

```diff
@@ -145,17 +145,22 @@ interface Option {
 }
 ```
 
-A `Reserved` represents an _expression_ with a _reserved_ _annotation_.
-The `sigil` corresponds to the starting sigil of the _reserved_.
+A `Reserved` represents an _expression_ with a _reserved_ or _private-use_ _annotation_.
+The `sigil` corresponds to the starting sigil of the _annotation_.
```

Comment:
I think this is incorrect.

Reserved represents a portion of the syntax that is wholly invalid--not permitted to be used--but which might become part of a future incarnation. If a part of reserved were used in a future incarnation, existing (pre the unreserving) implementations would continue with whatever behavior reserved provides. Probably this means, as you have it here, passing reserved gunk through the data model without processing.

Private use is different. Private use is a portion of the syntax that _is valid_, but which may not be functional in a given implementation. 

What I would do is define private use and reserved separately and use the same language where appropriate.

Also: note that private-use is a feature of our specification, which is why I put it before reserved in the text.

Reply:
Could you give an example of a data model use case where it would be useful for the _private-use_ and _reserved_ syntax rules to be represented by different data structures, rather than this single `Reserved`?

Given that supported private-use syntax would end up using a different interface altogether, I'm failing to come up with a realistic scenario where I'd like unsupported private-use and reserved to be handled differently. I suppose a linter error for a message including one or the other could use a slightly different text, but that can be detected from the `sigil`.